Foundation/FileManager+Win32.swift

```diff
@@ -581,8 +581,8 @@ extension FileManager {
         NSUnimplemented()
     }
 
-    internal func _lstatFile(atPath path: String, withFileSystemRepresentation fsRep: UnsafePointer<Int8>? = nil) throws -> stat {
-        let _fsRep: UnsafePointer<Int8>
+    internal func _lstatFile(atPath path: String, withFileSystemRepresentation fsRep: UnsafePointer<foundation_char_t>? = nil) throws -> stat {
```

Comment:
I don't really like the `foundation_char_t`.  Can we not do something like:

```Swift
#if os(Windows)
internal typealias NativeFileSystemCharacterType = WCHAR
#else
internal typealias NativeFileSystemCharacterType = CChar
#endif
```

I'm happy with alternate names such as `NativeFSRCharType` or something along those lines.